driver-core/src/main/com/mongodb/client/model/expressions/StringExpression.java

```diff
@@ -35,13 +35,25 @@ public interface StringExpression extends Expression {
 
     StringExpression substr(IntegerExpression start, IntegerExpression length);
 
-    default StringExpression substr(int start, int length) {
+    default StringExpression substr(final int start, final int length) {
         return this.substr(of(start), of(length));
     }
 
     StringExpression substrBytes(IntegerExpression start, IntegerExpression length);
 
-    default StringExpression substrBytes(int start, int length) {
+    default StringExpression substrBytes(final int start, final int length) {
         return this.substrBytes(of(start), of(length));
     }
+
+    /**
+     * parses to an underlying int64
+     * @return
+     */
+    IntegerExpression parseInteger();
```

Comment:
Why no `NumberExpression parseNumber()`

Reply:
$toInt is ranked #49, while toDouble and toDecimal (that would correspond to parseNumber) are ranked #59 and #72 (out of scope).

Reply:
Yuk. I thought we had decided to ignore the rankings in situations like this where there is a clear family of methods that go together.  If not, maybe we should.

Reply:
I did this for date, and included a ~dozen methods where only ~2 were in scope. I'm not opposed, but it will take some more time to implement and review. Let me know.

Reply:
How many are we talking about here.

Reply:
Agreed to add parseNumber.